NLSY/nlsy_impute.R

```diff
@@ -0,0 +1,160 @@
+#Packages
+librarian::shelf(dplyr, mice, car, corrplot, caret)
+
+# Read data
+nlsydf = readRDS(paste0(here::here(), "/NLSY/NLSY-college-finance.rds"))
+
+test <- nlsydf %>% #Add additional variables that should be considered during multiple imputation
+    select(id = "PUBID_1997",
+           mom_age_birth = "CV_BIO_MOM_AGE_YOUTH_1997", # numeric
+           age_resp =  "CV_AGE_12/31/96_1997",
+           hispanic = "KEY_ETHNICITY_1997", # binary
+           race = "KEY_RACE_1997", # categorical
+           mom_education = "CV_HGC_RES_MOM_1997", # categorical - highest grade completed in 1997 by mom
+           dad_education = "CV_HGC_RES_DAD_1997", # categorical - highest grade completed in 1997 by dad
+           savings = "P5-130_1997", #binary - yes/no retirement savings
+           home = "P5-101_1997", # binary - homeownership or renting
+           both_parents = "YOUTH_BOTHBIO.01_1997",
+           par1_deceased = "YOUTH_NONR1DEAD.01_1997",
+           par2_deceased = "YOUTH_NONR2DEAD.01_1997",
+           pincome = "CV_INCOME_GROSS_YR_1997",
+           pnetworth = "CV_HH_NET_WORTH_P_1997",
+           # Also, it seems that this represents the wealth of the household in which a respondent lives. That means that for a respondent who lives with only one parent, the net-worth variable represents the wealth of that parent.
+           # In the case of an independent respondent, contains respondent's, rather than parents', wealth.
+    ) %>%
+    mutate(mom_age = mom_age_birth + age_resp)
+
+# Making dummy variables
+test$hisp <- ifelse(test$hispanic == "Yes", 1, 0)
+test$ret_sav <- ifelse(test$savings == "YES", 1, 0)
```

Comment:
I think that the retirement savings variable has many missing values. Do we know how the mice package treats this type of predictor and whether we need to do anything special to prepare it for imputation?

Reply:
According to this vignette, when multiple variables are missing information, `mice` using a series of chained equations in which certain values are set randomly and then iteratively filled in: https://cran.r-project.org/web/packages/miceRanger/vignettes/miceAlgorithm.html. Looks like we don't need to do anything special beyond the pre-processing Sonia already did here

Reply:
Looking at the [original paper](https://www.jstatsoft.org/article/view/v045i03), it looks like we have the following options:
1. Run `mice` as is and the vignette above describes how the joint missingness problem is handled.
2. The authors suggesting dropping predictors that have too much missingness to have any useful information. It looks like retirement savings, home ownership, and father's education are all candidates. Intuition: If there are only a small number of observations where pincome *or* retirement savings is present, then omitting retirement savings will only have a small effect on the overall imputation model for pincome.
3. The authors suggest that if we really want to include a predictor in the imputation model but are concerned there is too much missingness as currently constructed, we can first run `mice` to impute *that* variable. My guess is this isn't drastically different than #1, but it could help to improve upon the initial randomization shown in the vignette diagram.

My inclination is to drop father's education (over 3000 missing obs) but then choose option 1 for remaining vars. But we can look into option 3 depending on how much time you think is worth spending on the imputation piece.